File: cpp/program/PrintedBook.cpp
#include <bits/stdc++.h>
#include "Book.cpp"

using namespace std;

// kelas book
class PrintedBook : public Book
{
private:
    // atribut-atribut private
    int pageCount;
    string coverType;

public:
    // konstruktor kosong
    PrintedBook()
    {
    }

    // konstruktor dengan parameter untuk mengisi atribut
    PrintedBook(string title, vector<Author> authorList, Publisher publisher, string ISBN, int pageCount, string coverType) : Book(title, authorList, publisher, ISBN)
    {
        this->pageCount = pageCount;
        this->coverType = coverType;
    }

    // setter pageCount
    void setPageCount(int pageCount)
    {
        this->pageCount = pageCount;
    }

    // getter pageCount
    int getPageCount()
    {
        return pageCount;
    }

    // setter coverType
    void setCoverType(string coverType)
    {
        this->coverType = coverType;
    }

    // getter coverType
    string getCoverType()
    {
        return coverType;
    }

    ~PrintedBook()
    {
    }
};

File: java/program/PrintedBook.java
import java.util.ArrayList;

public class PrintedBook extends Book {
    private int pageCount;
    private String coverType;

    // constructor dengan parameter
    PrintedBook(String title, ArrayList<Author> authorList, Publisher publisher, String ISBN, int pageCount, String coverType) {
        super(title, authorList, publisher, ISBN);
        this.pageCount = pageCount;
        this.coverType = coverType;
    }

    // getter dan setter untuk pageCount
    public void setPageCount(int pageCount) {
        this.pageCount = pageCount;
    }

    public int getPageCount() {
        return pageCount;
    }

    // getter dan setter untuk coverType
    public void setCoverType(String coverType) {
        this.coverType = coverType;
    }

    public String getCoverType() {
        return coverType;
    }
}
